src/zcbor_decode.c

```diff
@@ -920,6 +920,7 @@ bool zcbor_search_key_tstr_term(zcbor_state_t *state, char const *str, size_t ma
 }
 
 
+#ifndef ZCBOR_CANONICAL
```

Comment:
Why did we need to add this?

Reply:
Because some refactoring in `zcbor_skip_any` (now `traverse_ele ment`) means `array_end_expect` is only used when `ZCBOR_CANONICAL` is defined, so this was added to avoid a compiler warning.